CDS/cdsresponder/cdsresponder.rb

```diff
@@ -170,88 +194,105 @@ def ThreadFunc
           puts e.backtrace.inspect
           begin
             @notification_topic.publish({'status' => 'error', 'message' => e.message, 'trace' => e.backtrace}.to_json)
-          rescue Exception => e
-            puts "Error passing on error message: #{e.message}"
+          rescue Exception => not_excep
+            puts "Error passing on error message: #{not_excep.message}"
           end
 
         ensure
+          #if should_finish is set while messages still on the queue, terminate immediately by breaking out of the poll
+          #loop while not waiting for timeout
+          if @should_finish
+            puts "got should_finish signal"
+            @isexecuting = false
+            break
```

Comment:
the File.delete calls ensure that it cleans up after itself rather than splattering route and metadata files onto the disk then forgetting about them. Not a huge issue either way but i felt it's better not to leave stuff around